### DISCUSSION TITLE
Error: Function 'findloc' at (1) has no IMPLICIT type

### DISCUSSION THREAD

**Discussion by Blissful-Jasper** [NCAR/DART#676] [Q&A] [answered]
hi,all 

When I tried to compile the Lorenz 63 mode following the official website:

[https://docs.dart.ucar.edu/en/stable/guide/compiling-dart.html](url)

I found the following error.

```python
QTY_loc = findloc(specified_qtys, obs_qty)
          1
Error: Function 'findloc' at (1) has no IMPLICIT type
/Users/xpji/dart/DART/assimilation_code/modules/assimilation/algorithm_info_mod.f90:406.10:

QTY_loc = findloc(specified_qtys, qty)
          1
Error: Function 'findloc' at (1) has no IMPLICIT type
/Users/xpji/dart/DART/assimilation_code/modules/assimilation/algorithm_info_mod.f90:346.10:

QTY_loc = findloc(specified_qtys, obs_qty)
          1
Error: Function 'findloc' at (1) has no IMPLICIT type
make: *** [algorithm_info_mod.o] Error 1
````

![image](https://github.com/NCAR/DART/assets/60537770/2d494346-6232-46ef-bd9b-edbca51b04c3)

I'm not sure if this error is related to my netcdf path setting, hope to get your answer.

here's my mkmf.template:
```python
MPIFC = mpif90
MPILD = mpif90
FC = gfortran
LD = gfortran
NETCDF_F=/Software/netcdf-fortran-4.4.5
INCS = -I$(NETCDF_F)/include
LIBS = -L$(NETCDF_F)/lib  -lnetcdf 
FFLAGS  = -g -ffree-line-length-none  $(INCS)
LDFLAGS = $(FFLAGS) $(LIBS)
```

**Comment by hkershaw-brown** [comment, marked as answer]
Hi @xpji,

findloc is  a Fortran intrinsic introduced in the Fortran 2008 standard.   If you can, I'd recommend updating your compiler (gfortran version 9+)

If you cannot use a newer compiler, you can add this findloc routine to DART/assimilation_code/modules/assimilation/algorithm_info_mod.f90

https://github.com/NCAR/DART/issues/629#issuecomment-1914786208

Cheers,
Helen
